tests/substrate.rs

```diff
@@ -782,6 +786,16 @@ impl Runtime {
             .any(|account| account.contract.is_some() && account.address == address)
             .into())
     }
+
+    #[seal(0)]
+    fn set_code_hash(code_hash_ptr: u32) -> Result<u32, Trap> {
+        let hash = read_hash(mem, code_hash_ptr);
+        if let Some(code) = vm.blobs.iter().find(|code| code.hash == hash) {
+            vm.accounts[vm.account].contract.as_mut().unwrap().code = code.clone();
+            return Ok(0);
+        }
+        Ok(7) // ReturnCode::CodeNoteFound
```

Comment:
The code could come from the enum itself.

Reply:
I agree. That enum is defined in the contracts pallet. We could either pull that dependency in or maintain our own version. However there are many more places where it's done like that, and I'm heading off to decoded, so I'd rather clean this up another time.